samples/matter/common/src/bt_nus_service.h

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#pragma once
+
+#include "app_event.h"
+
+#include <platform/Zephyr/BLEAdvertisingArbiter.h>
+
+#include <bluetooth/services/nus.h>
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/bluetooth/conn.h>
+#include <zephyr/sys/ring_buffer.h>
+
+#include <lib/core/CHIPError.h>
+
+#include <list>
+
+class NUSService {
+public:
+	using CommandCallback = void (*)(void *context);
+	struct Command {
+		char command[CONFIG_CHIP_NUS_MAX_COMMAND_LEN];
+		size_t commandLen;
+		CommandCallback callback;
+		void *context;
+	};
+
+	/**
+	 * @brief Initialize BLE Lock service
+	 *
+	 * Initialize internal structures and register necessary commands in BLE service server.
+	 */
+	CHIP_ERROR Init(const char *name, size_t nameLen, uint8_t priority, uint16_t minInterval, uint16_t maxInterval);
+
+	/**
+	 * @brief Start BLE Lock service server
```

Comment:
Is this really limited to the BLE lock support? The API looks more generic that that and the user can register a custom command, right?